src/AddressDriver.sol

```diff
@@ -94,6 +94,27 @@ contract AddressDriver is Upgradeable, ERC2771Context {
     /// Positive to add funds to the drips balance, negative to remove them.
     /// @param newReceivers The list of the drips receivers of the sender to be set.
     /// Must be sorted by the receivers' addresses, deduplicated and without 0 amtPerSecs.
+    /// @param maxEndTip1 The first tip to calculate the maximum end time of drips
+    /// after the balance and the receivers list are updated.
+    /// Tips have no effect on the results of calling this function, but they can reduce
+    /// the range of timestamps when searching for the maximum end time, which saves gas.
+    /// Tips lower than the current timestamp are ignored,
+    /// and passing a zero is an easy way to not provide a tip.
+    /// It's acceptable to provide zero, one or two tips. The order of tips doesn't matter.
+    /// Tips are the most effective when one of them is lower than or equal to the maximum end time,
+    /// and the other one is strictly larger than the maximum end time,
+    /// the smaller the difference between them, the higher gas savings.
+    /// The savings are the highest possible when one of the tips
+    /// is equal to the maximum end time and the other one is larger by 1.
+    /// It's worth noting that the exact timestamp of the block in which this function is executed
+    /// may affect correctness of the tips, especially if they're precise.
+    /// Tips are used to divide the binary searched range of future timestamps into smaller parts,
+    /// so even inaccurate tips can be useful, and providing a single tip
+    /// or two tips that don't enclose the maximum end time can still save some gas.
+    /// Providing tips that don't limit the searched time range significantly
+    /// may cause slightly higher gas usage than not providing any tips.
+    /// @param maxEndTip2 The second tip to calculate the maximum end time of drips
+    /// after the balance and the receivers list are updated.
```

Comment:
Nice explanation. I would the information that the tips are acutally Unix timestamps.  Which is never said directly.

Reply:
Out of curiosity and because I would have called them `hints` instead of `tips`. 

> A tip is when you tell a person useful information, to help them. A hint is when you don't actually tell them, you say something different which might help them find that information themselves. 

For me it would be more a `hint` but I guess both is fine

Reply:
Added a note about Unix timestamps. Renaming tips to hints is a good idea, will apply in the next PR.